allennlp/modules/backbones/vilbert_backbone.py

```diff
@@ -123,7 +121,32 @@ def forward(
         # Shape: (batch_size, num_tokens)
         attention_mask = text["tokens"].get("mask")
 
-        # Shape: (batch_size, num_tokens, embedding_dim)
+        dimensions = box_features.shape
+        batch_size = dimensions[0]
+        feature_size = dimensions[-1]
+        rolled_dimensions = dimensions[1:-2]
+        rolled_dimensions_product = 1
+        for dim in rolled_dimensions:
+            rolled_dimensions_product *= dim
```

Comment:
I could have sworn there was a one-liner for this, but Guido said no: https://bugs.python.org/issue1093

Reply:
You could make it a one-liner in log space, but that would not help readability :-)

Reply:
That's true, but I think I'll keep the readable version :)

There actually is a one-liner now: https://docs.python.org/3/library/math.html (look for prod), but it's only in 3.8.